internal/stack/variants.go

```diff
@@ -6,20 +6,34 @@ package stack
 
 import (
 	"fmt"
-	"strings"
+
+	"github.com/Masterminds/semver/v3"
 )
 
 // stackVariantAsEnv function returns a stack variant based on the given stack version.
-// We identified two variants:
+// We identified three variants:
 // * default, covers all of 7.x branches
-// * 8x, supports different configuration options in Kibana
+// * 80, covers stack versions 8.0.0 to 8.1.x
+// * 8x, supports different configuration options in Kibana, covers stack versions 8.2.0+
 func stackVariantAsEnv(version string) string {
 	return fmt.Sprintf("STACK_VERSION_VARIANT=%s", selectStackVersion(version))
 }
 
 func selectStackVersion(version string) string {
-	if strings.HasPrefix(version, "8.") {
-		return "8x"
+	if v, err := semver.NewVersion(version); err == nil {
```

Comment:
Actually, you can convert these if-conditions into a `configurationVariantMap` (key: constraint, value: config variant).